tests/platform_helper/test_command_copilot.py

```diff
@@ -80,6 +82,217 @@
+        contents."""
+        # Arrange
+        mock_aws_client(mock_get_session)
+
+        addons_dir = FIXTURES_DIR / "make_addons"
+        fakefs.add_real_directory(
+            addons_dir / "config/copilot", read_only=False, target_path="copilot"
+        )
+        fakefs.add_real_file(
+            addons_dir / addon_file, read_only=False, target_path=ADDON_CONFIG_FILENAME
+        )
+        fakefs.add_real_directory(Path(addons_dir, "expected"), target_path="expected")
+
+        # make-addons will generate terraform compatible addons if it detects a ./terraform directory
+        fakefs.create_dir("./terraform")
+
+        # Act
+        result = CliRunner().invoke(copilot, ["make-addons"])
+
+        assert (
+            result.exit_code == 0
+        ), f"The exit code should have been 0 (success) but was {result.exit_code}"
+
+        assert ">>> Generating terraform compatbiele addons CloudFormation" in result.stdout
+
+        expected_service_files = [
+            Path("web/addons", filename) for filename in expected_service_addons
+        ]
+
+        for f in expected_service_files:
+            if "s3" in str(f):
+                # Use the terraform-* fixtures for s3
+                parts = (
+                    "expected",
+                    *f.parts[:-1],
+                    f"terraform-{f.name}",
+                )
+                expected_file = Path(*parts)
+            else:
+                expected_file = Path("expected", f)
+
+            expected = yaml.safe_load(expected_file.read_text())
+            actual = yaml.safe_load(Path("copilot", f).read_text())
+
+            assert sorted(expected) == sorted(
+                actual
+            ), f"The file {f} did not have the expected content"
+
+            assert actual == expected
+
+        assert not any(
+            Path("./copilot/environments/addons/").iterdir()
+        ), "./copilot/environments/addons/ should be empty"
+
+        env_override_files = setup_override_files_for_environments()
+        for file in env_override_files:
+            assert f"{file} created" in result.stdout
+
+        all_expected_files = expected_service_files + env_override_files
+
+        expected_svc_overrides_file = Path("expected/web/overrides/cfn.patches.yml").read_text()
+        actual_svc_overrides_file = Path("copilot/web/overrides/cfn.patches.yml").read_text()
+        assert actual_svc_overrides_file == expected_svc_overrides_file
+
+        copilot_dir = Path("copilot")
+        actual_files = [
+            Path(d, f).relative_to(copilot_dir)
+            for d, _, files in os.walk(copilot_dir)
+            for f in files
+        ]
+
+        assert (
+            len(actual_files) == len(all_expected_files) + 5
+        ), "The actual filecount should be expected files plus 3 initial manifest.yml and 1 override files"
+
+    # def test_warning_message_for_missing_kms_arn():
+    #     pass
```

Comment:
What's this?

Reply:
Ah, well spotted.  If we introspect aws and can't find a KMS alias for an s3 addon in a given environment, then 'kms-key-not-found' is rendered into the s3-policy extension for that environment.  

I was questioning whether we should warn that this happened, e.g 

"KMS key not found for my-s3-bucket in environment willg". 

At the same time: 

1) presumably trying to deploy a service before you've provisioned an environment is unlikely?
2) if you do happen to do this, then the the service will fail/roll back because of an an invalid ARN

We should probably warn anyway, so I'll add this.

Reply:
I thought about this, and I don't think we want it blocking.  Imagine our scenario with demodjango where there's a bajillion envs, and some are provisioned some aren't.  I've created at ticket to look at this as a tidyup/enhancement
https://uktrade.atlassian.net/wiki/spaces/DBTP/pages/3885269057/Terraform+Tuesday+and+Wednesday

Reply:
I hope we will be deploying a single environment at a time, so if the KMS key does not exist, it would be blocking one way or another.